test/cmocka/src/common_mocks.c

```diff
@@ -90,6 +90,17 @@ int WEAK memcpy_s(void *dest, size_t dest_size,
 	return 0;
 }
 
+int WEAK memset_s(void *dest, size_t size1,
+		  int c, size_t size2)
+{
+	if (!dest)
+		return -EINVAL;
+
+	memset_s(dest, c, size1);
```

Comment:
How does this even work? How does the linker distinguish between the weak symbol and the strong one you're calling here?

Reply:
it doesn't  - I pushed before I had merged the fix. Now fix is pushed.